src/client/opamCommands.ml

```diff
@@ -1292,26 +1287,20 @@ let install =
     if atoms = [] then
       (OpamConsole.msg "Nothing to do\n";
        OpamStd.Sys.exit_because `Success);
-    let deps_only =
-      if check then
-        let missing = OpamClient.check_installed st atoms in
-        if OpamPackage.Map.is_empty missing then
-          (OpamConsole.msg "All dependencies installed\n";
-           OpamStd.Sys.exit_because `Success)
-        else
-          (OpamConsole.error "Missing dependencies:\n%s"
-             (OpamStd.Format.itemize (fun (pkg, names) ->
-                  Printf.sprintf "%s: %s"
-                    (OpamConsole.colorise `underline (OpamPackage.to_string pkg))
-                    (OpamStd.List.concat_map ", " ~last_sep:" and/or "
-                       OpamPackage.Name.to_string
-                       (OpamPackage.Name.Set.elements names)))
-                 (OpamPackage.Map.bindings missing));
-           not (no_inst) && OpamConsole.confirm ~default:true "Install them?")
-      else deps_only
-    in
-    if check && not deps_only then
-      OpamStd.Sys.exit_because `False;
+    if check then
+      (let missing = OpamClient.check_installed st atoms in
```

Comment:
For consistency, and easier use in scripts, I would output in both cases:
- a readable message to stderr (`OpamConsole.errmsg`)
- the raw list of missing packages to stdout

Otherwise, this is perfect! Thanks :)

Reply:
With a raw list, we don't have the information of missing package -> atom package. I Suggest
```
p1.v1: m1, m2, m3
p2.v2: m4, m5
```
fi, current dispay is
```
- p1.v1: m1, m2 and/or m3
- p2.v2: m4 and/or m5
```

Reply:
note: lint err output is quieted to not interfere with std{err,out}